src/multerConfig.ts

```diff
@@ -0,0 +1,45 @@
+import { Request } from "express";
+import multer, { FileFilterCallback } from "multer";
+import path from "path"; // For generating unique filenames
+import fs from "fs";
+
+type FileNameCallback = (error: Error | null, filename: string) => void;
+// export const multerConfig = {
+//     storage: multer.memoryStorage(),
+//   limits: {
+//     fileSize: 5 * 1024 * 1024, // limit file size to 5MB
+//   },
+
+//     fileFilter: (req: Request, file: Express.Multer.File, cb: FileFilterCallback) => {       
+//         if (file.mimetype === "text/csv"){
+//             return cb(null, true); // Accept the file
+//         }
+//         // Reject other file types
+//         cb(null, false); // Reject the file
+//     }
+// };
+const uploadsDir = path.join(__dirname, 'uploads');
+if (!fs.existsSync(uploadsDir)) {
+    fs.mkdirSync(uploadsDir);
+}
+
+export const multerConfig = {
+  storage: multer.diskStorage({
```

Comment:
This should be memory storage as disk storage is saving them onto the local machine which will not work when uploaded to docker. Look at how Jack has done it in apply for role PR